src/client/proximaStreamClient.ts

```diff
@@ -12,13 +12,15 @@ export class ProximaStreamClient {
   private readonly registry: StreamRegistry;
   private readonly clients: Record<string, StreamDBConsumerClient>;
   private readonly offsetsCache: NodeCache;
+  private readonly apiKey?: string;
 
   public constructor(
     private readonly options: StreamClientOptions = {
       registry: new StreamRegistryClient(),
     }
   ) {
     this.registry = options.registry ?? new StreamRegistryClient();
+    this.apiKey = options.apiKey;
```

Comment:
If someone passed in a registry with api-key set, we should probably extract it from that registry.

Reply:
updated